lib/core.ps1

```diff
@@ -285,7 +285,47 @@ function isFileLocked([string]$path) {
 }
 
 function is_directory([String] $path) {
-    return (Test-Path $path) -and (Get-Item $path) -is [System.IO.DirectoryInfo]
+    return (Test-Path $path) -and (Get-Item -Path $path) -is [System.IO.DirectoryInfo]
+}
+
+function is_junction([String] $path) {
+    return (Test-Path $path) -and (Get-Item -Path $path).LinkType -eq 'Junction'
+}
+
+function create_junction([String] $link, [String] $target) {
+    if (!(Test-Path $link) -and (is_directory $target)) {
+        New-Item -ItemType Junction -Path (Split-Path -Path $link) -Name (Split-Path -Leaf $link) -Target $target | Out-Null
+        $dirInfo = New-Object System.IO.DirectoryInfo($link)
+        $dirInfo.Attributes = $dirInfo.Attributes -bor [System.IO.FileAttributes]::ReadOnly
+        return $true
+    }
+    return $false
+}
+
+function remove_junction([String] $link) {
+    if(is_junction $link) {
+        $dirInfo = New-Object System.IO.DirectoryInfo($link)
```

Comment:
`Remove-Item xxx -Recurse -Force` could remove read-only junction. So if we want to delete junction or hardlink, a single function that `Remove-Item xxx -Recurse -Force` is enough.